### PR TITLE
Battle 2k3: Implement CBA Afterimage

### PR DESCRIPTION
**Pull request by Ghabry**
~TODO: Figure out the transparency values.~

I made the API a bit more flexible: You can set how many afterimages you want :).

See #1269

Slightly exaggerated picture (manually set it to 10 images ;))

a-fs8

### CODE REVIEW

**Comment by Ghabry** [comment]
Figured out the transparency with a black-white-image and the 32 bit patch: Goes in steps of 64.

This is ready :)

**Review by fdelapena — APPROVED** [review]

File: src/scene_battle_rpg2k3.cpp [hunk 1]
```diff
@@ -2765,6 +2766,10 @@ void Scene_Battle_Rpg2k3::CBAInit() {
 			}
 		}
 	}
+
+	if (cba_action->GetCBAMovement() != lcf::rpg::BattlerAnimationItemSkill::Movement_none) {
+		sprite->SetAfterimageAmount(cba_action->GetWeaponAnimationData()->after_image == lcf::rpg::BattlerAnimationItemSkill::AfterImage_add ? 3 : 0);
+	}
```
Comment (fmatthew5876) [inline src/scene_battle_rpg2k3.cpp:2772]: When does this get reset to 0 after the movement is done?

What if you attack with after image and then..
* Escape?
* Row command?
* Change facing direction (but not movement) to attack an enemy on the other side?

Will those accidentally after-image as well?

Comment (Ghabry) [reply src/scene_battle_rpg2k3.cpp:2772]: The afterimage is removed when the battler is idling and all afterimage-frames have the same position.

The idling state is reached directly after the CBA movement is done, so 3 frames after the afterimages are gone. It is not possible to do any actions in this 3-frame window, as they are still part of the attack handling.

Though have to check if any event command can trigger it early. Could also set the amount of afterimages to "1" when the battle action starts. This way it is always reset.

**Comment by rueter37** [comment]
Works for me. :+1:

**Review by fmatthew5876 — APPROVED** [review]

File: src/scene_battle_rpg2k3.cpp (hunk 1 above)
Comment (Ghabry) [reply src/scene_battle_rpg2k3.cpp:2772]: Replaced the logic now with a fade-counter just for afterimage. This way it is ensured that the afterimage will always fade at the end of a CBA.

**Comment by rueter37** [comment]
Did some more testing and unfortunately found a bug: As soon as an actor uses a skill which involves CBA movement, the game crashes. Normal attacks are not affected by this bug. The crash does not happen in current master.

**Comment by rueter37** [comment]
@Ghabry: I have added a [commit](https://github.com/rueter37/player/tree/cba-afterimage) now which you can include in your PR now which should fix the afterimage issue with skills. There were more changes involved, because on normal attacks the afterimage setting is taken from the weapon animation data and on skills the afterimage setting is taken from the skill animation data.

**Comment by Ghabry** [comment]
Oh, thanks!
I totally forgot that Skills can also have CBA. This also means I never tested any of your CBA code with skills. I hope it works ^^

File: src/scene_battle_rpg2k3.cpp [hunk 2]
```diff
@@ -2768,7 +2768,7 @@ void Scene_Battle_Rpg2k3::CBAInit() {
 	}
 
 	if (cba_action->GetCBAMovement() != lcf::rpg::BattlerAnimationItemSkill::Movement_none) {
-		sprite->SetAfterimageAmount(cba_action->GetWeaponAnimationData()->after_image == lcf::rpg::BattlerAnimationItemSkill::AfterImage_add ? 3 : 0);
+		sprite->SetAfterimageAmount(cba_action->GetCBAAfterImage() == lcf::rpg::BattlerAnimationItemSkill::AfterImage_add ? 3 : 0);
```
Comment (Ghabry) [inline src/scene_battle_rpg2k3.cpp:2771]: I just noticed: This is incorrect in the liblcf enum. It is an Afterimage, not an AfterImage.
Will submit a patch to liblcf to fix this. Good moment right now, will not break anythign except the PR :)

**Comment by rueter37** [comment]
> I totally forgot that Skills can also have CBA. This also means I never tested any of your CBA code with skills. I hope it works ^^

CBA in skills are restricted to the options "Pose", "Movement" and "Afterimage". The first two are already implemented since #2503 so the skill CBA code works. Normal attacks have more CBA options available.

**Comment by rueter37** [comment]
@Ghabry: I have updated my [commit](https://github.com/rueter37/player/tree/cba-afterimage) now according to the recent liblcf changes.

**Comment by Ghabry** [comment]
@rueter37 thanks! I replaced yoru commit now.

**Pull request merged**